### ISSUE TITLE
[question] How one can pass data from tag to the title?

### ISSUE THREAD

**Issue by matkoniecz** [westnordost/StreetComplete#582]
As I understand https://github.com/westnordost/StreetComplete/commit/4500cb6d08740cbc0fcda73bb475917241f56ff3 makes impossible to pass arbitrary data to the title.

It seems to break some quests that are at this moment purely hypothetical in StreetComplete, but I had them in my fork.

For example displaying text of fixme tag (or other tags) as the quest question no longer appears to be possible. Real affected code includes https://github.com/matkoniecz/StreetComplete/blob/fork/app/src/main/java/de/westnordost/streetcomplete/quests/show_fixme/ShowFixmeForm.java#L29 (fixme quest) or https://github.com/matkoniecz/StreetComplete/blob/fork/app/src/main/java/de/westnordost/streetcomplete/quests/road_surface/showInvalidSurfaceForm.java#L23 (displaying unexpected surface values).

Is there some way now to achieve this effect? Or is it necessary for me to either remove this type of quest or make fork deeper and include my own code to handle quests requiring freeform tag title?

**Comment by matkoniecz** [comment]
In case that for some reason this quest type (freeform title based on tag values) is not supposed to ever appear in SC feel free to close (after all, problem with changes upstream is an obvious consequence of maintaining a fork).

**Comment by rugk** [comment]
> (after all, problem with changes upstream is an obvious consequence of maintaining a fork).

Better try to avoid forks and get everything into upstream. StreetComplete still needs to (and can) grow and we don't need multiple versions of the same app with just slight variations.

**Comment by westnordost** [comment]
You are correct that this is not designed to work that way anymore. You'd have to tear quite a bit of code up to enable it to be the way it worked for you before.
Otherwise, see @rugk. I don't understand why you maintain a fork just for yourself if your work could have 5000x the impact it does now (there are about 5000 StreetComplete users right now).

**Comment by matkoniecz** [comment]
> I don't understand why you maintain a fork just for yourself

Reasons include

- quests that should be opt-in (blocked by #35)
- testing things
- customizing things (fork started as change to quest priorities as opening hours quest in my case is boring and with little value, as there is not enough mappers in my city to cover changing opening hours of all POIs)
- doing things that were explicitly rejected in SC
- because it is much easier to implement some stuff "it works" than "it works properly and may be a PR"
- and finally, some stuff that should be turned into a PR sooner so I will not spend time on chasing upstream changes

**Comment by krzyk** [comment]
I have my own fork just like @matkoniecz because of the above reasons (I've extended few existing quests to cover more objects, and added my own quest). And also because I'm not a newbie and I'm OK if there are quests that I can't solve, or have some corner cases that are not targeted for basic SC users. 

Some changes that I made (that are specific to my usecase, and wouldn't be OK for general public):
- I prefer to see all the quests possible in my vicinity, not t wait until I solve more important quests, so my screen is stormed with quests at the place (I increased the quest per km2, and added more quests to download at once)
- I want to add roofs even for cases where roof:levels = 0 (some roof levels are too small to be considered a level), and before overpass notice that I added roof:levels - again solve as many quests in given place as I possibly can
- I want to add surface to footways (most of the surfaces from highways apply here)
- I want to add more lit=yes/no to highways and footways

Adding tags using SC on the go is so convenient that I prefer to use it instead of other editors (and keeping a survey journal). So I don't think those changes apply to general SC public (maybe with exclusion of seeing more quests and the roof one), or need more work (in case of footway surface).
